lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2109,7 +2109,23 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
   }
   case OpCopyLogical: {
     SPIRVCopyLogical *CL = static_cast<SPIRVCopyLogical *>(BV);
-    return mapValue(BV, transSPIRVBuiltinFromInst(CL, BB));
+
+    auto *SrcTy = transType(CL->getOperand()->getType());
+    auto *SrcAI = new AllocaInst(SrcTy, 0, "", BB);
+    new StoreInst(transValue(CL->getOperand(), F, BB), SrcAI, BB);
```

Comment:
Perhaps add a comment explaining why we need a store first, instead of using the result of `transValue(CL->getOperand(), F, BB)` directly?

Reply:
Do you have a specific suggestion?  I'm open to improving this.  I'm not aware of other viable options, though...

Reply:
To me it feels like there is some redundancy when doing a `store` and then a `memcpy`.  Thinking about this again, could we not avoid the `memcpy` altogether, and pretty much follow `OpCopyObject` above?  The only difference would be that the type on the `load` is different from the type on the `store`; we have both types already available (as `SrcTy` and `DstTy` in your current patch).

Reply:
Interesting - I thought it would have been a problem to store as one type and load as a different type, but it seemed to work just fine.  Have a look at the latest version - much simpler now!

(I'll edit the PR description also, since there is no longer a memcpy involved.)

Reply:
LGTM, thanks!